src/Image.php

```diff
@@ -25,13 +25,16 @@ class Image
 	/** @var string */
 	public $name;
 
-	/** @var ImageNameScript */
+	/** @var ImageNameScript|null */
 	private $script;
 
 	/** @var bool */
 	private $friendly_url = false;
 
-	public function __construct(bool $friendly_url, string $data_dir, string $data_path, string $identifier, ?Traversable $props = null)
+	/**
+	 * @param iterable<string, bool|string|ImageNameScript|null>|null $props
```

Comment:
Please don't use generics syntax until PSR-5 generics are accepted. There is missing support in a lot of tools for that.

Reply:
slevomat/cs should support this already, phpstorm supports this already

Reply:
I tested it few days ago with phpstorm 2018.3 and it was only available for writing generics, no typehint when using variables with generics.

Reply:
no typehint? what do you mean by that? there is no support in php itself.